src/System.Private.CoreLib/shared/System/Threading/Tasks/ValueTask.cs

```diff
@@ -415,7 +416,7 @@ internal void ThrowIfCompletedUnsuccessfully()
         /// <summary>null if <see cref="_result"/> has the result, otherwise a <see cref="Task{TResult}"/> or a <see cref="IValueTaskSource{TResult}"/>.</summary>
         internal readonly object? _obj;
         /// <summary>The result to be used if the operation completed successfully synchronously.</summary>
-        internal readonly TResult _result;
+        [AllowNull] internal readonly TResult _result;
```

Comment:
Should its property also have `MaybeNull`?

Reply:
Ah, we shouldn't as that will be stated by the generic type parameter used to create the instance.

Reply:
Right.